### DISCUSSION TITLE
RPi3/4 64bit Release i20231022 Duplicated DUID

### DISCUSSION THREAD

**Discussion by markblinkhorn** [rern/rAudio#1488] [Bug Reports]
OK, this has had me running around chasing my tail and doubting my sanity for the last few days. 

I installed 2 new RPi 3B+ with the 20231022 64bit image and my DHCP server kept handing out the same (reserved) IP address to both Pi machines. I had already modified the hostnames on both.

Cut a long story short - it turns out that the file /etc/machine.id in the 20231022 64bit image already has content. Systemd-networkd uses this file to create a DUID to offer to the DHCP server in exchange for an IP address. Both machines end up with the same 'unique' DUID and so they both get offered the same IP address!

The workaround: Before the SD card is used to boot a Pi for the first time the /etc/machine.id file needs to be emptied.  After the image has been written to the SD card it must be mounted on a Linux PC to edit the mounted /etc/machine.id. (I used nano on a PC running Manjaro, but most anything should work). Clear all of the content of the file and write a zero length file back to the SD card. Check the file is read only (chmod 444) before you unmount the SD card.  I have NOT checked the other images to see if they have the same issue.

I hope this helps someone and I am now happy that my lack of sanity is beyond doubt.

EDIT: for /etc/machine.id read /etc/machine-id throughout. Thanks @ceejayemm

**Comment by ceejayemm** [comment]
@markblinkhorn 
Thanks, that probably explains the odd DHCP messages I have been seeing but thought it was something I was doing as I have been playing around with my Pihole settings for DHCP..

With existing build can I simply clear the same file and reboot each machine of would it be better to reflash the SD cards of each and start from scratch ?

Chris

**Comment by ceejayemm** [comment]
@markblinkhorn 
BTW - On my running systems the file is /etc/machine-id (not machine.id).

Chris

**Reply by markblinkhorn** [reply]
Thanks Chris - it's a typo. Mea culpa.

I *guess* you can clear the content on a running system. I haven't tried it yet. But I will...

Mark

**Reply by markblinkhorn** [reply]
@ceejayemm 
Yes, it works ;-)
